src/simple_lottery/init.rs

```diff
@@ -0,0 +1,94 @@
+//! Parameter derivation for simple lottery.
+
+use super::params::Params;
+use super::setup::Setup;
+
+use std::f64::consts::LN_2;
+
+/// Calculates Setup parameters.
+pub fn make_setup(params: &Params) -> Setup {
+    let ratio = params.set_size as f64 / params.lower_bound as f64;
+
+    // Execute binary search for the optimal value of `rs`.
+
+    let mut l: f64 = 0.0;
```

Comment:
I know it's a pedantic personal preference :)  but I prefer giving slightly more descriptive names than `l`, `m`, `r`: 
`l` --> `left`
`r` --> `middle`
... 
specifically, I'll rename `rs` and `rc` to something more descriptive as it's not very intuitive what the abbreviations mean.

Reply:
renamed `rs`, `rc` for consistency